core/src/services/onedrive/backend.rs

```diff
@@ -80,13 +80,19 @@ impl Access for OnedriveBackend {
     }
 
     async fn read(&self, path: &str, args: OpRead) -> Result<(RpRead, Self::Reader)> {
-        let response = self.core.onedrive_get_content(path, args.range()).await?;
+        let response = self
+            .core
+            .onedrive_get_content(path, args.range(), args.if_none_match())
+            .await?;
 
         let status = response.status();
         match status {
             StatusCode::OK | StatusCode::PARTIAL_CONTENT => {
                 Ok((RpRead::default(), response.into_body()))
             }
+            StatusCode::NOT_MODIFIED => {
```

Comment:
Maybe we can handle this in `parse_error`?

Reply:
Ok.